os/Xtranssock.c

```diff
@@ -1180,7 +1178,7 @@ static int _XSERVTransSocketBytesReadable (
 	(void *) ciptr, ciptr->fd, (void *) pend);
 #ifdef WIN32
     {
-	int ret = ioctlsocket ((SOCKET) ciptr->fd, FIONREAD, (u_long *) pend);
+	int ret = ossock_ioctl((SOCKET) ciptr->fd, FIONREAD, (u_long *) pend);
```

Comment:
Do we need to keep the whole `#ifdef WIN32` construct here? `ossock_ioctl` already checks for the error code. I'm not sure if casting the last argument in the call (in the original code) makes any difference.